jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdConfiguration.java

```diff
@@ -69,60 +71,94 @@ public OpenIdConfiguration(String provider, String clientId, String clientSecret
      * @param clientSecret The client secret known only by the Client and the Authorization Server.
+     * @param httpClient The {@link HttpClient} instance to use.
      */
-    public OpenIdConfiguration(String issuer, String authorizationEndpoint, String tokenEndpoint, String clientId, String clientSecret)
+    public OpenIdConfiguration(String issuer, String authorizationEndpoint, String tokenEndpoint,
+                               String clientId, String clientSecret, HttpClient httpClient)
     {
         this.issuer = issuer;
         this.clientId = clientId;
         this.clientSecret = clientSecret;
+        this.authEndpoint = authorizationEndpoint;
+        this.tokenEndpoint = tokenEndpoint;
+        this.httpClient = httpClient;
 
-        if (issuer == null)
-            throw new IllegalArgumentException("Provider was not configured");
+        if (this.issuer == null)
+            throw new IllegalArgumentException("Issuer was not configured");
 
-        if (tokenEndpoint == null || authorizationEndpoint == null)
+        addBean(this.httpClient);
+    }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        super.doStart();
+
+        if (authEndpoint == null || tokenEndpoint == null)
         {
-            Map<String, Object> discoveryDocument = fetchOpenIdConnectMetadata(issuer);
+            Map<String, Object> discoveryDocument = fetchOpenIdConnectMetadata(issuer, httpClient);
 
-            this.authEndpoint = (String)discoveryDocument.get("authorization_endpoint");
-            if (this.authEndpoint == null)
+            authEndpoint = (String)discoveryDocument.get("authorization_endpoint");
+            if (authEndpoint == null)
                 throw new IllegalArgumentException("authorization_endpoint");
 
-            this.tokenEndpoint = (String)discoveryDocument.get("token_endpoint");
-            if (this.tokenEndpoint == null)
+            tokenEndpoint = (String)discoveryDocument.get("token_endpoint");
+            if (tokenEndpoint == null)
                 throw new IllegalArgumentException("token_endpoint");
 
             if (!Objects.equals(discoveryDocument.get("issuer"), issuer))
-                LOG.warn("The provider in the metadata is not correct.");
-        }
-        else
-        {
-            this.authEndpoint = authorizationEndpoint;
-            this.tokenEndpoint = tokenEndpoint;
+                throw new IllegalArgumentException("The provider in the metadata is not correct.");
         }
     }
 
-    private static Map<String, Object> fetchOpenIdConnectMetadata(String provider)
+    private static HttpClient newHttpClient()
+    {
+        SslContextFactory.Client sslContextFactory = new SslContextFactory.Client(false);
+        sslContextFactory.setEndpointIdentificationAlgorithm("https");
+        HttpClient client = new HttpClient(sslContextFactory);
+        client.setFollowRedirects(false);
```

Comment:
I would remove this. On the surface, it should work and seems prudent. We have nothing in any spec telling us that redirection won't happen though. So, I think it could cause interop issues.